packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -553,6 +554,16 @@ class SliverGeometry extends Diagnosticable {
   /// By definition, this cannot be less than [paintExtent].
   final double maxPaintExtent;
 
+  /// The maximum extent by which this sliver can reduce the area in which
+  /// content can scroll if the sliver were pinned at the edge.
+  ///
+  /// Slivers, that never get pinned at the edge, should return zero.
+  ///
+  /// A pinned app bar is an example for a sliver that would use this setting:
+  /// When the app bar is pinned to the top, the area in which content can
+  /// actually scroll is reduced by the height of the app bar.
+  final double maxPinnedExtent;
```

Comment:
why not minPaintExtent?

Reply:
I feel like `minPaintExtent` doesn't describe the same thing. I'd for example say that the floating app bar has a minPaintExtent of zero because it can be completely invisible. However, if it is visible, it might be blocking other slivers -- and that's the extent I need to know. If this sliver is pinned at the top what's the most it can cover? Basically what I need is `maxExtentByWhichSliverWillEverReduceTheScrollableAreaIfPinnedAtEdge` -- that seemed too long though and `maxPinnedExtent` was all I could come up with. Happy to go with other suggestions, though :)

Reply:
What does the floating app bar report for maxPinnedExtent when it's entirely off-screen?

Reply:
It reports `maxExtent` because that's how much space it could potentially occupy in the pinned position if it decides to become visible due to the scrolling action triggered by this call to ensureVisible.

Reply:
So this is really neither a "min" nor a "max", it's more like "protectedExtent" or some such, the region that ensureVisible should avoid. Maybe "deflectExtent", or "scrollObstructionExtent", or "scrollingOcclusionExtent" or some such.

Reply:
It's still a max because I need the maximum amount that this sliver is ever going to obstruct scrolling by.

Reply:
I'm leaning towards renaming this to `maxScrollObstructionExtent`: The maximum extent by which this sliver might obstruct scrolling when pinned to the edge of the scroll view.